File: pinyougou_parent/pinyougou_shop_web/src/main/webapp/js/service/goodsService.js
//服务层
app.service('goodsService',function($http){
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../goods/findAll.do');		
	}
	//分页||条件查询
        this.search = function (page, size, entity) {
            return $http.post("../goods/search.do?page=" + page + "&size=" + size, entity);
        }
        //增加
        this.add = function (entity) {
            return $http.post("../goods/add.do", entity);
        }
        //修改
        this.update = function (entity) {
            return $http.post("../goods/update.do", entity);
        }
        //查询实体
        this.findById = function (id) {
            return $http.get("../goods/findById.do?id=" + id);
        }
        //删除
        this.del = function (ids) {
            return $http.get("../goods/delete.do?ids=" + ids);
        }
});

File: pinyougou_parent/pinyougou_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/GoodsService.java
package com.pinyougou.sellergoods.service;
import java.util.List;
import com.pinyougou.pojo.TbGoods;

import com.pinyougou.pojo.TbItem;
import com.pinyougou.vo.GoodsVo;
import entity.PageResult;
import entity.Result;

/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface GoodsService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TbGoods> findAll();

	
	
	/**
	 * 增加
     * @param goods
     */
	public Result add(GoodsVo goods);
	
	
	/**
	 * 修改
	 * @param goods
	 */
	public void update(GoodsVo goods);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public GoodsVo findById(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(TbGoods goods, int pageNum, int pageSize);

    void updateStatus(Long[] ids, String status);
	List<TbItem> findItemListByGoodsIdandStatus(Long[] goodsIds, String status);
}
